lib/SILGen/SILGenDecl.cpp

```diff
@@ -1330,16 +1330,30 @@ void SILGenFunction::emitStmtCondition(StmtCondition Cond, JumpDest FalseDest,
     case StmtConditionElement::CK_Availability:
       // Check the running OS version to determine whether it is in the range
       // specified by elt.
-      VersionRange OSVersion = elt.getAvailability()->getAvailableRange();
+      PoundAvailableInfo *availability = elt.getAvailability();
+      VersionRange OSVersion = availability->getAvailableRange();
       assert(!OSVersion.isEmpty());
 
       if (OSVersion.isAll()) {
         // If there's no check for the current platform, this condition is
-        // trivially true.
+        // trivially true (or false, for unavailability).
         SILType i1 = SILType::getBuiltinIntegerType(1, getASTContext());
-        booleanTestValue = B.createIntegerLiteral(loc, i1, true);
+        bool value = !availability->getIsUnavailability();
+        booleanTestValue = B.createIntegerLiteral(loc, i1, value);
       } else {
         booleanTestValue = emitOSVersionRangeCheck(loc, OSVersion);
+        if (availability->getIsUnavailability()) {
+          // If this is an unavailability check, invert the result
+          // by emitting a call to Builtin.cmp_eq_Int1(lhs, 0).
+          SILType i1 = SILType::getBuiltinIntegerType(1, getASTContext());
+          SILValue zero = B.createIntegerLiteral(loc, i1, false);
+          booleanTestValue =
+            B.createBuiltinBinaryFunction(loc,
+                                          "cmp_eq",
+                                          i1,
+                                          i1,
+                                          {booleanTestValue, zero});
+        }
```

Comment:
I was unsure if this is the best way of inverting the result. IIRC the `!` operator XORs the result, which should be faster but I couldn't figure out to emit a call to it here.

Reply:
I don't know if there's a better way to do it but you can use `createIntegerLiteral(loc, i1, 1);` and `xor` instead of `cmp_eq`, I think that should do the trick.

Reply:
I got it now! I don't remember what the problem was before, I probably just confused it with something else 😄